includes/discovery/sensors/current/adva_fsp150.inc.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * LibreNMS - ADVA device support - Current
+ *
+ * @category   Network_Monitoring
+ * @package    LibreNMS
+ * @subpackage ADVA device support
+ * @author     Christoph Zilian <czilian@hotmail.com>
+ * @license    http://gnu.org/copyleft/gpl.html GNU GPL
+ * @link       https://github.com/librenms/librenms/
+
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ **/
+
+// ******************************************
+// ***** Sensors for ADVA FSP150EG-X Chassis
+// ******************************************
+
+    // Define Sensors and Limits
+    $sensors_adva = array
+                (
+                array(
+                        'sensor_name'     => 'psuOutputCurrent',
+                        'sensor_oid'      => '.1.3.6.1.4.1.2544.1.12.3.1.4.1.8')
+                );
+
+    $multiplier = 1;
+    $divisor    = 1000;
+    $sensorType = 'adva_fsp150';
+    
+    foreach (array_keys($pre_cache['adva_fsp150']) as $index) {
+        foreach ($sensors_adva as $entry) {
+            $sensor_name = $entry['sensor_name'];
+            if ($pre_cache['adva_fsp150'][$index][$sensor_name]) {
+                $oid          = $entry['sensor_oid'].".".$index;
+                $rrd_filename = $pre_cache['adva_fsp150'][$index]['slotCardUnitName']."-".$pre_cache['adva_fsp150'][$index]['slotIndex'];
```

Comment:
I really think this is over complicating things. `$rrd_filename` should be named `$index` anyway as that's what the variable is but the setting of how it is is overkill and should be simplified to `psuOutputCurrent.$index`

This can be looked up in the db if more info is needed to work out what that index references.